app/src/org/commcare/gis/MapboxLocationPickerActivity.kt

```diff
@@ -0,0 +1,213 @@
+import org.commcare.activities.components.FormEntryConstants
+import org.commcare.dalvik.R
+import org.commcare.utils.ConnectivityStatus
+import org.commcare.utils.GeoUtils
+import org.commcare.views.widgets.GeoPointWidget
+
+class MapboxLocationPickerActivity : BaseMapboxActivity() {
+
+    companion object {
+        const val MARKER_ICON_IMAGE_ID = "marker_icon_image"
+    }
+
+    private lateinit var viewModel: MapboxLocationPickerViewModel
+    private var loadedStyle: Style? = null
+    private val mapStyles = arrayOf(
+            Style.MAPBOX_STREETS,
+            Style.SATELLITE,
+            Style.SATELLITE_STREETS,
+            Style.OUTDOORS
+    )
+    private var currentMapStyleIndex = 0
+    private var cameraPosition: CameraPosition? = null
+    private var symbolManager: SymbolManager? = null
+    private var symbol: Symbol? = null
+    private val mapClickListener = MapboxMap.OnMapClickListener { point ->
+        // Add marker.
+        updateMarker(point)
+        viewModel.reverseGeocode(point)
+        true
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        viewModel = ViewModelProvider(this, ViewModelProvider.AndroidViewModelFactory.getInstance(application))
+                .get(MapboxLocationPickerViewModel::class.java)
+        attachListener()
+
+        // Check extras
+        if (intent.hasExtra(GeoPointWidget.LOCATION)) {
+            val inputLocation = intent.getDoubleArrayExtra(GeoPointWidget.LOCATION)!!
+            cameraPosition = CameraPosition.Builder()
+                    .target(LatLng(inputLocation[0], inputLocation[1]))
+                    .zoom(10.0)
+                    .tilt(20.0)
+                    .build()
+        }
+        mapView.showCurrentLocationBtn(false)
+    }
+
+    private fun attachListener() {
+        confirm_location_button.visibility = if (inViewMode()) {
+            View.GONE
+        } else {
+            View.VISIBLE
+        }
+        confirm_location_button.setOnClickListener {
+            // Use the map target's coordinates to make a reverse geocoding search
+            Intent().apply {
+                putExtra(FormEntryConstants.LOCATION_RESULT, GeoUtils.locationToString(viewModel.getLocation()))
+            }.also {
+                setResult(RESULT_OK, it)
+                finish()
+            }
+        }
+        cancel_button.setOnClickListener {
+            finish()
+        }
+        switch_map.setOnClickListener {
+            if (ConnectivityStatus.isNetworkAvailable(this)) {
+                currentMapStyleIndex = (currentMapStyleIndex + 1) % mapStyles.size
+                val style = Style.Builder()
+                        .fromUri(mapStyles[currentMapStyleIndex])
+                        .withImage(MARKER_ICON_IMAGE_ID, ContextCompat.getDrawable(this, R.drawable.marker)!!)
+                map.setStyle(style) {
```

Comment:
should we be calling `setMapStyle()` instead ? I see that we are not setting `loadedStyle` here unlike ``setMapStyle()` which can cause issues.

Reply:
I removed `loadedStyle` variable since it was unused.